pyxem/signals/vdf_image.py

```diff
@@ -30,3 +38,109 @@ def __init__(self, *args, **kwargs):
+            them to be considered as separate grains.
+        min_size : float
+            Grains with size (i.e. total number of pixels) below
+            min_size are discarded.
+        max_size : float
+            Grains with size (i.e. total number of pixels) above
+            max_size are discarded.
+        max_number_of_grains : int
+            Maximum number of grains included in the returned separated
+            grains. If it is exceeded, those with highest peak
+            intensities will be returned.
+        marker_radius : float
+            If 1 or larger, each marker for watershed is expanded to a disk
+            of radius marker_radius. marker_radius should not exceed
+            2*min_distance.
+        threshold: bool
+            If True, a mask is calculated by thresholding the VDF image
+            by the Li threshold method in scikit-image. If False
+            (default), the mask is the boolean VDF image.
+        exclude_border : int or True, optional
+            If non-zero integer, peaks within a distance of
+            exclude_border from the boarder will be discarded. If True,
+            peaks at or closer than min_distance of the boarder, will be
+            discarded.
+
+        References
+        ----------
+        [1] http://scikit-image.org/docs/dev/auto_examples/segmentation/
+            plot_watershed.html
+        [2] http://scikit-image.org/docs/dev/auto_examples/xx_applications/
+            plot_coins_segmentation.html#sphx-glr-auto-examples-xx-
+            applications-plot-coins-segmentation-py
+
+        Returns
+        -------
+        vdfsegs : VDFSegment
+            VDFSegment object containing segments (i.e. grains) of
+            single virtual dark field images with corresponding vectors.
+        """
+        vdfs = self.copy()
+        vectors = self.vectors.data
+
+        # TODO : Add aperture radius as an attribute of VDFImage?
+
+        # Create an array of length equal to the number of vectors where each
+        # element is a np.object with shape (n: number of segments for this
+        # VDFImage, VDFImage size x, VDFImage size y).
+        vdfsegs = np.array(vdfs.map(
+            separate_watershed, show_progressbar=True, inplace=False,
+            min_distance=min_distance, min_size=min_size, max_size=max_size,
+            max_number_of_grains=max_number_of_grains,
+            marker_radius=marker_radius, threshold=threshold,
+            exclude_border=exclude_border), dtype=np.object)
+
+        segments, vectors_of_segments = [], []
+        for i, vector in zip(np.arange(vectors.size), vectors):
+            segments = np.append(segments, vdfsegs[i])
+            num_segs = np.shape(vdfsegs[i])[0]
+            vectors_of_segments = np.append(
+                vectors_of_segments, np.broadcast_to(vector, (num_segs, 2)))
+
+        if segments.dtype == object:
+            segments = segments.astype(float)
```

Comment:
Why is this needed? type checking is usually a bad thing. If needed, needs testing.

Reply:
I've removed this as necessary type checking.